enacts/calc.py

```diff
@@ -1,27 +1,101 @@
 import numpy as np
 import pandas as pd
 import xarray as xr
+import datetime
 
 # Date Reading functions
-def read_zarr_data(zarr_path):
-    """Read and return data in zarr format.
+
+np.random.seed(123)
+
+
+def read_enacts(variable, ds_conf):
+    """ Read ENACTS data
 
     Parameters
-    ------
-    zarr_path : str
-        String of path to zarr folder.
+    ----------
+    variable : str
+        string representing ENACTS variable ("precip", "tmin" or "tmax")
+    ds_conf : dict
+        dictionary indicating ENACTS zarr path (see config)
+    
     Returns
     -------
-    zarr_data : Dataset
-        Data from zarr folder as multidimensional xarray dataset.
+        `xr.DataArray` of ENACTS `variable` at `time_res` time steps
+    
     See Also
     --------
-    Notes
-     -----
+    xr.open_zarr
     """
-    zarr_data = xr.open_zarr(zarr_path)
-    return zarr_data
+    try:
+        data_path = ds_conf['vars'][variable][1]
+        if data_path is None:
+            data_path = ds_conf['vars'][variable][0]
+        zarr_path = f"{ds_conf['zarr_path']}{data_path}"
+        var_name = ds_conf['vars'][variable][2]
+    except KeyError as e:
+        raise Exception(f'configuration of data reading raised error {e}')
```

Comment:
What do you intend to accomplish by catching and reraising, as opposed to just letting the KeyError be raised?